src/main/java/seedu/address/ui/WalletStatisticsPanel.java

```diff
@@ -0,0 +1,163 @@
+import javafx.collections.transformation.FilteredList;
+import javafx.fxml.FXML;
+import javafx.scene.chart.PieChart;
+import javafx.scene.control.Label;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.ReadOnlyWallet;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.transaction.Budget;
+import seedu.address.model.transaction.Date;
+import seedu.address.model.transaction.Expense;
+import seedu.address.model.transaction.Transaction;
+
+/**
+ * Ui panel containing the statistics of the Wallet.
+ */
+public class WalletStatisticsPanel extends UiPart<Region> {
+
+    private static final String FXML = "WalletStatisticsPanel.fxml";
+
+    private static final String BUDGET_NOT_SET = "No budget set!";
+    private static final String UNDER_BUDGET = "You are $%.2f under your budget! Good job!";
+    private static final String OVER_BUDGET = "You are $%.2f over your budget! Oh no!";
+
+    private static final String OVER_BUDGET_CLASS = "budget-over";
+    private static final String UNDER_BUDGET_CLASS = "budget-under";
+
+    private final Logger logger = LogsCenter.getLogger(WalletStatisticsPanel.class);
+
+    @FXML
+    private PieChart expenditurePieChart;
+
+    @FXML
+    private Label currentMonthYearLabel;
+
+    @FXML
+    private Label budgetRemainingLabel;
+
+    @FXML
+    private Label budgetOverUnderLabel;
+
+    private ObservableList<Transaction> walletTransactionList;
+
+    private ReadOnlyWallet wallet;
+
+    public WalletStatisticsPanel(ReadOnlyWallet wallet, ObservableList<Transaction> transactionList) {
+        super(FXML);
+        update(wallet, transactionList);
+    }
+
+    /**
+     * Updates the statistics displayed with the modified {@code wallet} and {@code transactionList}.
+     */
+    public void update(ReadOnlyWallet wallet, ObservableList<Transaction> transactionList) {
+        this.wallet = wallet;
+        this.walletTransactionList = transactionList;
+
+        resetStyles();
+        setCurrentMonthYear();
+        populatePieChart();
+        updateBudgetRemaining();
+    }
+
+    /**
+     * Updates the current month / year label to the current date.
+     */
+    private void setCurrentMonthYear() {
+        Date currDate = Date.getDefault();
+        currentMonthYearLabel.setText(String.format("%s %s", currDate.getMonth(), currDate.getYear()));
+    }
+
+    /**
+     * Populates the pie chart with the data provided.
+     */
+    private void populatePieChart() {
+        ObservableList<PieChart.Data> pieChartData = FXCollections.observableArrayList();
+
+        Date currDate = Date.getDefault();
+        List<Tag> tagList = new ArrayList<Tag>();
+        List<Double> totalAmountList = new ArrayList<Double>();
+
+        for (Transaction t : walletTransactionList) {
+            if (!tagList.contains(t.getTag())) {
+                tagList.add(t.getTag());
+            }
+        }
+        for (Tag tag : tagList) {
```

Comment:
I think to avoid multiple iterations can just build a hashmap of tag -> amount with one pass.

Reply:
I'm going to leave it as it is as the later part for the legend requires iteration through the lists in order. When I add the items and set the data, the Legend follows that order and it will be difficult to get the items in order for the Legend of the PieChart to work.